src/components/forms/VpnServerForm.js

```diff
@@ -0,0 +1,424 @@
+            onToggle={() => this.handleToggle("enable_split_tunneling")}
+            className="fieldWidthSmaller"
+          />
+          {/* client_idle_timeout input */}
+          <NumberInput
+            id={
+              this.props.data.name + "-vpn-server-client-idle-timeout-seconds"
+            }
+            name={
+              this.props.data.name + "-vpn-server-client-idle-timeout-seconds"
+            }
+            placeholder="600"
+            label="Client Idle Timeout (In Seconds)"
+            allowEmpty={true}
+            value={this.state.client_idle_timeout || ""}
+            step={1}
+            onChange={this.handleNumberInputChange}
+            hideSteppers={true}
+            min={0}
+            max={28800}
+            invalid={isRangeInvalid(this.state.client_idle_timeout, 0, 28800)}
+            invalidText="Must be a whole number between 0 and 28800."
+            className="fieldWidth leftTextAlign"
+          />
+        </IcseFormGroup>
+        <IcseFormGroup>
+          {/* text area for client dns server IPs */}
+          <TextArea
+            className="textInputMedium"
+            id={this.props.data.name + "-vpn-server-client-dns-server-ips"}
+            labelText="Client DNS Server IPs"
+            placeholder={
+              this.state.client_dns_server_ips || "X.X.X.X, X.X.X.X/X, ..."
+            }
+            value={String(this.state.client_dns_server_ips)}
+            onChange={this.handleAllowedIps}
+            invalid={isIpStringInvalid(this.state.client_dns_server_ips)}
+            invalidText="Please enter a comma separated list of IP addresses."
+            helperText="Enter a comma separated list of IP addresses."
+          />
+        </IcseFormGroup>
+        {/* show routes if not modal */}
+        {this.props.isModal === false && (
+          <IcseFormTemplate
+            name="Routes"
+            subHeading
+            addText="Create a Route"
+            arrayData={this.props.data.routes}
+            innerForm={VpnServerRouteForm}
+            disableSave={this.props.vpnServerRouteProps.disableSave}
+            onDelete={this.props.vpnServerRouteProps.onDelete}
+            onSave={this.props.vpnServerRouteProps.onSave}
+            onSubmit={this.props.vpnServerRouteProps.onSubmit}
+            propsMatchState={this.props.propsMatchState}
+            innerFormProps={{ ...routeProps }}
+            hideAbout
+            toggleFormProps={{
+              hideName: true,
+              submissionFieldName: "routes",
+              disableSave: this.props.vpnServerRouteProps.disableSave,
+              type: "formInSubForm",
+            }}
+          />
+        )}
+      </>
+    );
+  }
+}
+
+VpnServerForm.defaultProps = {
+  data: {
+    name: "",
+    certificate_crn: "",
+    method: "",
+    client_ca_crn: "",
+    client_ip_pool: "",
+    enable_split_tunneling: false,
+    client_idle_timeout: "",
```

Comment:
should be a number

getting warning
```
 Warning: Failed prop type: Invalid prop `data.client_idle_timeout` of type `string` supplied to `VpnServerForm`, expected `number`.
 
```

Reply:
No, this is getting a warning and the input this goes to is a number input. I think error here comes from the story having a string passed in as props, but regardless, we shouldn't be passing string to a number input

Reply:
@jvallexm what do you think about this? `handleNumberInputChange` runs `parseInt` anyways so if a user enters something in either `port` or `client_idle_timeout` it will get converted anyways.

Reply:
if you are looking for feedback here @Ay1man2 please ping me directly, i have **all notifications turned off**